lib/pool.js

```diff
@@ -184,14 +199,53 @@ Pool.prototype.execute = function(sql, values, cb) {
   });
 };
 
+Pool.prototype._startExpiredTimer = function() {
+  if (!this._expiredTimer) {
+    this._expiredTimer = Timers.setInterval(
+        Pool.prototype._closeIdleConnections.bind(this),
+        1000
+    );
+  }
+};
+
+Pool.prototype._stopExpiredTimer = function() {
+  if (this._expiredTimer) {
+      Timers.clearInterval(this._expiredTimer);
+      this._expiredTimer = null;
+  }
+};
+
+Pool.prototype._closeIdleConnections = function() {
+  var now = Date.now();
+  var timeout = this.config.idleTimeout * 1000;
+  while (this._extraConnections.length) {
+    var conn = this._extraConnections.peek();
+
+    if (now > conn._lastReleased + timeout) {
+      // This connection has been unused for longer than the timeout
+      this._extraConnections.shift();
+      conn.destroy();
```

Comment:
this should be `.end()`. First, it's better to do `COM_QUIT` command and let server know that we wont to close connection. Second, I can add multiple queries to connection and release it without waiting for last command to complete. In that case `.destroy()` would kill all commands that hasn't been started